x-pack/plugins/endpoint/public/applications/endpoint/index.tsx

```diff
@@ -52,6 +66,101 @@ interface RouterProps {
   store: Store;
   coreStart: CoreStart;
   depsStart: EndpointPluginStartDependencies;
+  ingestManager: IngestManagerSetup;
+}
+
+const IsAppUnavailable: React.FunctionComponent<{
+  ingestManager: IngestManagerSetup;
+  setIsAppUnavailable: React.Dispatch<React.SetStateAction<boolean>>;
+}> = ({ ingestManager, setIsAppUnavailable }) => {
+  (async () => {
+    if (await isIngestManagerInitialized(ingestManager)) {
+      setIsAppUnavailable(false);
+      return null;
+    }
+  })();
+
+  const [hasIngestSetupFinished, setHasIngestSetupFinished] = React.useState(false);
+  const [isIngestInitSuccessful, setIsIngestInitSuccessful] = React.useState(false);
+  const [ingestInitError, setIngestInitError] = React.useState<Error | undefined>(undefined);
+  React.useEffect(() => {
+    ingestManager
```

Comment:
I got a comment further below that might impact this call here. TLDR == should we use our own http client and do a REST call instead of importing the service from ingest.